tests/encryption/test_intransit_encryption_sanity.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.resources.storage_cluster import (
+    in_transit_encryption_verification,
+    set_in_transit_encryption,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_intransit_encryption_notset,
+    tier1,
+    skipif_ocs_version,
+)
+
+log = logging.getLogger(__name__)
+
+
+@skipif_intransit_encryption_notset
+class TestInTransitEncryptionSanity:
+    def teardown(self):
+        set_in_transit_encryption()
+
+    @tier1
+    @skipif_ocs_version("<4.12")
+    @pytest.mark.polarion_id("OCS-4861")
+    def test_intransit_encryption_enable_disable_statetransition(self):
+        """
+        1. Verify in-transit Encryption is Enable on setup.
+        2. Disable Encryption
+        3. Verify in-transit encryption configuration is removed.
+        4. Enable encryption Again and verify it.
+        5. Verify in-transit encryption config is exists.
+        """
+        log.info("Verifying the in-transit encryption is enable on setup.")
+        assert in_transit_encryption_verification()
+
+        log.info("Disabling the in-transit encryption.")
+        set_in_transit_encryption(enabled=False)
```

Comment:
there should be a fixture with a finalizer that in case of a test failure, would bring the cluster back to how it was before the test

Reply:
Added Finalizer statement in test step.